### ISSUE TITLE
Differences in Preact.Children API and React.Children API

### ISSUE THREAD

**Issue by mq2thez** [preactjs/preact#2644]
Hey friends!

I noticed a small difference in how `React.Children` is implemented between `preact/compat` and React.

`preact/compat` children: https://github.com/preactjs/preact/blob/master/compat/src/Children.js#L3
`react` children: https://github.com/facebook/react/blob/121af3143c7164a196420b080cb31d0ef07d5dff/packages/react/src/ReactChildren.js#L223

The different here being that the React implementation allows for a `context` arg, while the `preact/compat` version does not. While it's easily fixable on our end, it seemed like the kind of thing which might lead to incompatibilities in library code.

**Comment by ParSal123** [comment]
Is this issue still open?

**Comment by developit** [comment]
Yes. It's unlikely to be fixed, as no issues have ever been opened citing this as a real-world compatibility issue.

**Comment by ParSal123** [comment]
Does this mean I can't work on it? I saw the "beginner-friendly" tag.
